### PR TITLE
Adds with_counts to dpy.

### PR DESCRIPTION
**Pull request by JDJGInc**
## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/client.py [hunk 1]
```diff
@@ -1325,6 +1325,12 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
 
             ``guild_id`` parameter is now positional-only.
 
+        with_counts: Optional[:class:`bool`]
+            Whether to include count information in the guild. This fills the
+            :attr:`.Guild.approximate_member_count` and :attr:`.Guild.approximate_presence_count`
+            attributes without needing any privileged intents. Defaults to ``True``.
+            .. versionadded:: 2.0
```
Comment (StockerMC) [inline discord/client.py:1332]:
```suggestion

            .. versionadded:: 2.0
```

File: discord/client.py [hunk 2]
```diff
@@ -1307,7 +1307,7 @@ async def fetch_template(self, code: Union[Template, str]) -> Template:
         data = await self.http.get_template(code)
         return Template(data=data, state=self._connection)  # type: ignore
 
-    async def fetch_guild(self, guild_id: int, /) -> Guild:
+    async def fetch_guild(self, guild_id: int, /, *, with_counts: Optional[bool] = True) -> Guild:
```
Comment (Gobot1234) [inline discord/client.py:1310]: This type hint is wrong, it should only allow bool

Comment (JDJGInc) [reply discord/client.py:1310]: it's an Optional Kwarg.
I don't want to break it.

Comment (StockerMC) [reply discord/client.py:1310]: `Optional[T]` means `T | None`,  but it only accepts a `bool`

Comment (JDJGInc) [reply discord/client.py:1310]: do you mean with_counts: bool = True ?

Comment (Gobot1234) [reply discord/client.py:1310]: Yes

Comment (StockerMC) [reply discord/client.py:1310]: Yes, that's correct

Comment (JDJGInc) [reply discord/client.py]: There you go

File: discord/guild.py [hunk 3]
```diff
@@ -255,6 +255,19 @@ class Guild(Hashable):
...
+        using :meth:`Client.fetch_guild` with ``with_counts=True``.
+        .. versionadded:: 2.0
+
+    approximate_presence_count: Optional[:class:`int`]
+        The approximate number of members currently active in the guild.
+        This includes idle, dnd, online, and invisible members. Offline members are excluded.
+        This is ``None`` unless the guild is obtained using :meth:`Client.fetch_guild`
+        with ``with_counts=True``.
+    
+        .. versionchanged:: 2.0
```
Comment (Rapptz) [inline discord/guild.py:270]:
```suggestion
    approximate_member_count: Optional[:class:`int`]
        The approximate number of members in the guild. This is ``None`` unless the guild is obtained
        using :meth:`Client.fetch_guild` with ``with_counts=True``.

        .. versionadded:: 2.0
    approximate_presence_count: Optional[:class:`int`]
        The approximate number of members currently active in the guild.
        Offline members are excluded. This is ``None`` unless the guild is obtained using 
        :meth:`Client.fetch_guild` with ``with_counts=True``.
    
        .. versionchanged:: 2.0
```

File: discord/guild.py [hunk 4]
```diff
@@ -463,6 +478,8 @@ def _from_data(self, guild: GuildPayload) -> None:
         self._public_updates_channel_id: Optional[int] = utils._get_as_snowflake(guild, 'public_updates_channel_id')
         self.nsfw_level: NSFWLevel = try_enum(NSFWLevel, guild.get('nsfw_level', 0))
         self.mfa_level: MFALevel = try_enum(MFALevel, guild.get('mfa_level', 0))
+        self.approximate_presence_count = guild.get("approximate_presence_count")
+        self.approximate_member_count = guild.get("approximate_member_count")
```
Comment (Rapptz) [inline discord/guild.py:482]:
```suggestion
        self.approximate_presence_count: Optional[int] = guild.get('approximate_presence_count')
        self.approximate_member_count: Optional[int] = guild.get('approximate_member_count')
```

File: discord/http.py [hunk 5]
```diff
@@ -1075,8 +1075,9 @@ def get_guilds(
     def leave_guild(self, guild_id: Snowflake) -> Response[None]:
         return self.request(Route('DELETE', '/users/@me/guilds/{guild_id}', guild_id=guild_id))
 
-    def get_guild(self, guild_id: Snowflake) -> Response[guild.Guild]:
-        return self.request(Route('GET', '/guilds/{guild_id}', guild_id=guild_id))
+    def get_guild(self, guild_id: Snowflake, *, with_counts: bool = True) -> Response[guild.Guild]:
+        params = {"with_counts": int(with_counts)}
```
Comment (Rapptz) [inline discord/http.py:1079]:
```suggestion
        params = {'with_counts': int(with_counts)}
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
This library uses single quotes not double quotes.

Did you check whether the appropriate API typings in `discord.types` need to be updated?

File: discord/guild.py (hunk 3 above)
Comment (dolfies) [reply discord/guild.py:270]: IMO it would be better to just use the existing member_count property instead of a new attribute.

Comment (JDJGInc) [reply discord/guild.py]: If people want me to do that I can do that right now.

**Comment by JDJGInc** [comment]
my current code for with_counts works with slash commands and normally with commands.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**